radixdlt-core/radixdlt/src/main/java/com/radixdlt/sync/LocalSyncService.java

```diff
@@ -456,23 +458,34 @@ private SyncingState updateTargetIfNeeded(
 			) > 0;
 
 		if (isNewerState) {
-			return currentState
+			final var newState = currentState
 				.withTargetHeader(header)
 				.withCandidatePeers(peers);
+			this.updateSyncTargetDiffCounter(newState);
```

Comment:
If you make `updateSyncTargetDiffCounter` return input, you'll be allowed to simplify code down to 
`return this.updateSyncTargetDiffCounter(newState);`.